docs/migrations/01241.md

```diff
@@ -0,0 +1,103 @@
+# Purpose
+
+Updates Styletron to V5 and edits style types. Previously to this change, `styled` calls were not
+type checked. Removes usage of `$ref` now that Styletron has support for `ref` forwarding.
+
+See the example below for how this change will affect your code. Type definitions of props passed to
+components created with `styled` will be moved out of the style function, and to between `styled` and
+arguments. View information of Flow Generics [here](https://flow.org/en/docs/types/generics/#toc-syntax-of-generics).
+Theme no longer needs to be explictly typed in props. The `styled` function provides that.
+
+```diff
+import {styled} from 'baseui';
+
+- type PropsT = {$active: boolean, $theme: ThemeT};
+- const MyStyledComponent = styled('div', (props: PropsT) => {
+-  return {
+-    backgroundColor: props.$active ? props.$theme.colors.primary400 : props.$theme.colors.negative400,
+-  };
+- });
+
++ type PropsT = {$active: boolean};
++ const MyStyledComponent = styled<PropsT>('div', props => {
```

Comment:
Specify the type of the object: `(props: PropsT)`

Reply:
it's not necessary since `PropsT` is provided as a generic